### PR TITLE
Feature/upgrade service layer

### PR DESCRIPTION
**Pull request by daskyrk**
## What type of PR is this?

- [ ] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [x] Refactor
- [ ] Style
- [ ] Chore

## What this PR does / why we need it:
Add api-creator to generate service and related hook automatically by api config.
Migrate domain manage to new api-creator.

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [ ] Yes(screenshot is required)
- [x] No

### CODE REVIEW

File: core/src/service/api-creator.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  success: boolean;
+  err: {
+    msg: string;
+    code: string;
+    ctx: null | object;
+  };
+  userInfo?: Record<string, object>;
+}
+
+const noop = (d: any) => {};
```
Comment (McDaddy) [inline core/src/service/api-creator.ts:152]: optional call  instead of noop

File: core/src/service/api-creator.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+ */
+export const genRequest = function <T extends FN>(apiConfig: APIConfig) {
+  const { api, headers } = apiConfig;
+  let [method, path] = api.split('@');
+  if (!path) {
+    path = method;
+    method = 'get';
+  }
+
+  return (params: CallParams & Merge<Parameters<T>[0], {}>) => {
```
Comment (McDaddy) [inline core/src/service/api-creator.ts:84]: no need declare Merge 
`params: CallParams & Parameters<T>[0] extends never ? {} : Parameters<T>[0]`

File: core/src/service/api-creator.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    }
+    return axios({
+      method: method as any,
+      url: generatePath(path, pathParams),
+      headers: !isEmpty(headers) ? headers : $headers,
+      params: bodyOrQuery,
+      paramsSerializer: (p: Record<string, string>) => qs.stringify(p),
+      responseType: isDownload ? 'blob' : 'json',
+      data: bodyData,
+    }).then((res) => res.data) as unknown as Promise<BODY<ReturnType<T>>>;
```
Comment (McDaddy) [inline core/src/service/api-creator.ts:105]: async/await instead of promise

File: core/src/service/api-creator.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+  };
+};
+
+export const apiDataStore = createStore({
+  name: 'api_data',
```
Comment (McDaddy) [inline core/src/service/api-creator.ts:110]: do we have name standard for names like store name。 `_` looks a little wried

File: core/src/service/api-creator.ts (hunk 1 above)
Comment (daskyrk) [reply core/src/service/api-creator.ts:152]: It's used before replaced with real hook setState.

File: core/src/service/api-creator.ts (hunk 3 above)
Comment (daskyrk) [reply core/src/service/api-creator.ts:105]: promise is fine here, no need to wrap with generator in bundled code.

File: core/src/service/api-creator.ts (hunk 4 above)
Comment (daskyrk) [reply core/src/service/api-creator.ts]: Nope, have changed to apiData

File: core/src/service/api-creator.ts (hunk 2 above)
Comment (daskyrk) [reply core/src/service/api-creator.ts:84]: well, Merge is used to extract the inner property out, so we can see the structure directly. 
image

**Review by McDaddy — APPROVED** [review]

**Pull request merged**